packages/frontend/src/App.tsx

```diff
@@ -1,64 +1,37 @@
 import { IndexedDBStorageAdapter } from "@automerge/automerge-repo-storage-indexeddb";
-import { type FirebaseOptions, initializeApp } from "firebase/app";
 import invariant from "tiny-invariant";
 import * as uuid from "uuid";
 
 import { MultiProvider } from "@solid-primitives/context";
-import { Navigate, type RouteDefinition, type RouteSectionProps, Router } from "@solidjs/router";
-import { FirebaseProvider } from "solid-firebase";
-import { ErrorBoundary, Show, createResource, createSignal, lazy } from "solid-js";
+import {
+    Navigate,
+    type RouteDefinition,
+    type RouteSectionProps,
+    Router,
+} from "@solidjs/router";
+
+import { ErrorBoundary, Show, createResource, lazy } from "solid-js";
 
-import Dialog, { Content, Portal } from "@corvu/dialog";
-import { getAuth, signOut } from "firebase/auth";
-import { type Api, ApiContext, createRpcClient, useApi } from "./api";
 import { helpRoutes } from "./help/routes";
 import { createModel } from "./model/document";
 import { PageContainer } from "./page/page_container";
 import { TheoryLibraryContext, stdTheories } from "./stdlib";
 import { ErrorBoundaryDialog } from "./util/errors";
+import { Api, ApiContext, useApi } from "./api";
 
-const serverUrl = import.meta.env.VITE_SERVER_URL;
 const repoUrl = import.meta.env.VITE_AUTOMERGE_REPO_URL;
-const firebaseOptions = JSON.parse(import.meta.env.VITE_FIREBASE_OPTIONS) as FirebaseOptions;
 
 const Root = (props: RouteSectionProps<unknown>) => {
-    invariant(serverUrl, "Must set environment variable VITE_SERVER_URL");
     invariant(repoUrl, "Must set environment variable VITE_AUTOMERGE_REPO_URL");
-    const serverHost = new URL(serverUrl).host;
-
-    const firebaseApp = initializeApp(firebaseOptions);
-    const rpc = createRpcClient(serverUrl, firebaseApp);
 
     const repo = new Repo({
         storage: new IndexedDBStorageAdapter("catcolab"),
         network: [new BrowserWebSocketClientAdapter(repoUrl)],
     });
 
-    const api: Api = { serverHost, rpc, repo };
-
-    const [isSessionInvalid] = createResource(
-        async () => {
-            const result = await rpc.validate_session.query();
-            if (result.tag === "Err") {
-                await signOut(getAuth(firebaseApp));
-
-                // Why this needs to be a separate modal:
-                // We cannot automatically reload the page because a bug in validate_session might
-                // trigger an infinite reload loop, so the reload must be user-triggered. Although
-                // ErrorBoundary might seem like the natural place to handle this, it only catches the
-                // first error, and there's no guarantee that an error from validate_session will be the
-                // first one encountered.
-                return true;
-            }
-
-            return false;
-        },
-        {
-            initialValue: false,
-        },
-    );
+    const api: Api = { repo };
```

Comment:
This was where I "cut in" -- by removing the serverHost and rpc I was able to isolate which parts of the application wouldn't work in Patchwork. Obviously we'll want to discuss if and how to recover those capabilities.